<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NativeAppPage</class>
 <widget class="QWidget" name="NativeAppPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>511</width>
    <height>580</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Executable</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Project Target:</string>
        </property>
        <property name="buddy">
         <cstring>projectTargetRadio</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QRadioButton" name="projectTargetRadio">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="ProjectTargetsComboBox" name="projectTarget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Executable:</string>
        </property>
        <property name="buddy">
         <cstring>executableRadio</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QRadioButton" name="executableRadio">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KUrlRequester" name="executablePath">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="placeholderText">
           <string>Enter the executable name or absolute path to an executable file</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Behavior</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Arguments:</string>
        </property>
        <property name="buddy">
         <cstring>arguments</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="arguments">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter arguments to give to the executable.&lt;br/&gt;You can pass arguments containing space characters by putting them in double-quotes.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="placeholderText">
         <string>Enter arguments to give to the executable</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Working Directory:</string>
        </property>
        <property name="buddy">
         <cstring>workingDirectory</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="KUrlRequester" name="workingDirectory">
        <property name="toolTip">
         <string>Select a working directory for the executable</string>
        </property>
        <property name="placeholderText">
         <string>Select a working directory for the executable</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Environment:</string>
        </property>
        <property name="buddy">
         <cstring>environment</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="KDevelop::EnvironmentSelectionWidget" name="environment">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Select an environment to be used</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KDevelop::EnvironmentConfigureButton" name="configureEnvironment"/>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>12</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="runInTerminal">
          <property name="toolTip">
           <string>&lt;p&gt;By default applications will be run in the background and only their output will be displayed in a toolview. This makes it impossible to interact with applications requiring user input from a terminal emulator. To run such applications, you should use an external terminal.&lt;/p&gt;</string>
          </property>
          <property name="text">
           <string>Use External Terminal:</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="1">
       <widget class="KComboBox" name="terminal">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>&lt;p&gt;Defines the command to execute the external terminal emulator. Use the following placeholders:&lt;/p&gt;
&lt;dl&gt;
&lt;dt&gt;&lt;code&gt;%exe&lt;/code&gt;&lt;/dt&gt;
&lt;dd&gt;The path to the executable selected above.&lt;/dd&gt;
&lt;dt&gt;&lt;code&gt;%workdir&lt;/code&gt;&lt;/dt&gt;
&lt;dd&gt;The path to the working directory selected above.&lt;/dd&gt;
&lt;/dl&gt;
&lt;p&gt;The arguments defined above will get appended to this command.&lt;/p&gt;</string>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <item>
         <property name="text">
          <string notr="true">konsole --noclose --workdir %workdir -e %exe</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">xterm -hold -e %exe</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">gnome-terminal -e %exe</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Dependencies</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Action:</string>
        </property>
        <property name="buddy">
         <cstring>dependencyAction</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KComboBox" name="dependencyAction">
        <property name="toolTip">
         <string>Specifies the action to take for the dependencies before starting the executable.</string>
        </property>
        <property name="whatsThis">
         <string>The selected action will be run before the executable is started.  This allows there to be parts of a project, upon which the executable does not directly depend, to be built and/or installed before running the application.</string>
        </property>
        <property name="frame">
         <bool>true</bool>
        </property>
        <item>
         <property name="text">
          <string>Do Nothing</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string comment="@action">Build</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string comment="@action">Build and Install</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Targets:</string>
        </property>
        <property name="buddy">
         <cstring>targetDependency</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="ProjectItemLineEdit" name="targetDependency">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Enter a dependency to add to the list</string>
          </property>
          <property name="placeholderText">
           <string>Enter a dependency to add to the list</string>
          </property>
          <property name="showClearButton" stdset="0">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="browseProject">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="addDependency">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Adds the listed target to the dependency list.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="0" rowspan="4" colspan="2">
         <widget class="QListWidget" name="dependencies">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>List of indirect dependent targets.</string>
          </property>
          <property name="whatsThis">
           <string>This list should contain targets that the application does not directly depend on, but for which an action needs to be taken before running the application.</string>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="uniformItemSizes">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="removeDependency">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Removes the selected dependencies from the list.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="moveDepUp">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Move a dependency up in the list.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QPushButton" name="moveDepDown">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Moves the selected dependency down in the list.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
  <customwidget>
   <class>QListWidget</class>
   <extends>QListWidget</extends>
   <header>klistwidget.h</header>
  </customwidget>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>KDevelop::EnvironmentSelectionWidget</class>
   <extends>KComboBox</extends>
   <header>util/environmentselectionwidget.h</header>
  </customwidget>
  <customwidget>
   <class>ProjectItemLineEdit</class>
   <extends>QLineEdit</extends>
   <header>project/projectitemlineedit.h</header>
  </customwidget>
  <customwidget>
   <class>ProjectTargetsComboBox</class>
   <extends>QComboBox</extends>
   <header>projecttargetscombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KDevelop::EnvironmentConfigureButton</class>
   <extends>QToolButton</extends>
   <header>util/environmentconfigurebutton.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>executablePath</tabstop>
  <tabstop>arguments</tabstop>
  <tabstop>workingDirectory</tabstop>
  <tabstop>environment</tabstop>
  <tabstop>targetDependency</tabstop>
  <tabstop>addDependency</tabstop>
  <tabstop>dependencies</tabstop>
  <tabstop>removeDependency</tabstop>
  <tabstop>moveDepUp</tabstop>
  <tabstop>moveDepDown</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>projectTargetRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>executablePath</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>executableRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>executablePath</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runInTerminal</sender>
   <signal>toggled(bool)</signal>
   <receiver>terminal</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>227</y>
    </hint>
    <hint type="destinationlabel">
     <x>206</x>
     <y>226</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>projectTargetRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>projectTarget</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>152</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>executableRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>projectTarget</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>310</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
